mlflow/dspy/wrapper.py

```diff
@@ -0,0 +1,125 @@
+import json
+from typing import Any, Dict, Optional
+
+from mlflow.exceptions import INVALID_PARAMETER_VALUE, MlflowException
+from mlflow.protos.databricks_pb2 import (
+    INVALID_PARAMETER_VALUE,
+)
+from mlflow.pyfunc import PythonModel
+
+
+class DspyModelWrapper(PythonModel):
+    """MLflow PyFunc wrapper class for Dspy models.
+
+    This wrapper serves two purposes:
+        - It stores the Dspy model along with dspy global settings, which are required for seamless
+            saving and loading.
+        - It provides a `predict` method so that it can be loaded as an MLflow pyfunc, which is
+            used at serving time.
+    """
+
+    def __init__(
+        self,
+        model: "dspy.Module",  # noqa: F821
```

Comment:
You need to import `dspy.Module` using `if TYPE_CHECKING` like this:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from dspy import Module
```

Reply:
I tried that before, but the VScode still has the visual error:
<img width="861" alt="image" src="https://github.com/user-attachments/assets/bdcbc4d8-df34-4933-b8dc-7af3b6b115c7">

That's actually one reason I hate type hints -  annotating an arg as a dep's class type is so much pain.

Reply:
that's because you do not import `dspy`.

```python
from foo import bar

foo()  # <- foo is undefined
```

Reply:
lol my bad